uat/testing-features/src/main/java/com/aws/greengrass/TestUtils.java

```diff
@@ -0,0 +1,66 @@
+package com.aws.greengrass;
+
+//import com.aws.greengrass.platforms.Platform;
+//import com.aws.iot.evergreen.common.Constants;
+//
+//import io.netty.util.internal.StringUtil;
+import com.github.jknack.handlebars.internal.lang3.StringUtils;
+import lombok.extern.log4j.Log4j2;
+
+import java.util.function.Predicate;
+
+import static com.aws.greengrass.Constants.DEFAULT_GENERIC_POLLING_TIMEOUT_MILLIS;
+
+
+@Log4j2
+public final class TestUtils {
+
+    private TestUtils() {
+    }
+
+    /**
+     * check if it will fullfill the condition in given time
+     *
+     * @param condition the condition used to test the reslt
+     * @param optional for specify the time.
+     * @return
+     * @throws InterruptedException
+     */
+    public static boolean eventuallyTrue(Predicate<Void> condition, long... optional) throws InterruptedException {
+        long timeoutInMillis = optional.length >= 1 ? optional[0] : DEFAULT_GENERIC_POLLING_TIMEOUT_MILLIS;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;optional&#39; (index: 1 | type: long[). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;condition&#39;. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).